ITR/data/base_providers.py

```diff
@@ -358,44 +388,63 @@ def get_company_intensity_and_production_at_base_year(self, company_ids: List[st
-                self.column_config.COMPANY_ID)
+        excluded_cols = ['projected_targets', 'projected_intensities', 'historic_data']
+        df = pd.DataFrame.from_records(
+            [ICompanyData.parse_obj({k:v for k, v in c.dict().items() if k not in excluded_cols}).dict()
+             for c in self.get_company_data(company_ids)]).set_index(self.column_config.COMPANY_ID)
+        return df
 
-    def get_company_projected_trajectories(self, company_ids: List[str]) -> pd.DataFrame:
+    def get_company_projected_trajectories(self, company_ids: List[str], year=None) -> pd.DataFrame:
         """
         :param company_ids: A list of company IDs
-        :return: A pandas DataFrame with projected intensity trajectories per company, indexed by company_id
-        """
-        trajectory_list = [self._convert_projections_to_series(c, self.column_config.PROJECTED_EI, EScope.S1S2)
-                           for c in self.get_company_data(company_ids)]
-        if trajectory_list:
-            with warnings.catch_warnings():
-                # pd.DataFrame.__init__ (in pandas/core/frame.py) ignores the beautiful dtype information adorning the pd.Series list elements we are providing.  Sad!
-                warnings.simplefilter("ignore")
-                return pd.DataFrame(trajectory_list)
+        :param year: values for a specific year, or all years if None
+        :return: A pandas DataFrame with projected intensity trajectories per company, indexed by company_id and scope
+        """
+        projected_trajectories = [(c.company_id, scope, getattr(c.projected_intensities, scope.name).projections)
+                                  # FIXME: we should make _companies a dict so we can look things up rather than searching every time!
+                                  for c in self._companies for scope in EScope.get_result_scopes()
+                                  if c.company_id in company_ids
+                                  if getattr(c.projected_intensities, scope.name)]
+        if projected_trajectories:
+            if year:
+                company_ids, scopes, values = list(map(list, zip(*[(pt[0], pt[1], [yvp.value for yvp in pt[2] if yvp.year==year][0]) for pt in projected_trajectories])))
+                index=pd.MultiIndex.from_tuples(zip(company_ids, scopes), names=["company_id", "scope"])
+                return pd.Series(data=values, index=index)
+            else:
+                company_ids, scopes, yvp_dicts = list(map(list, zip(*[(pt[0], pt[1], {yvp.year:yvp.value for yvp in pt[2]}) for pt in projected_trajectories])))
+                index=pd.MultiIndex.from_tuples(zip(company_ids, scopes), names=["company_id", "scope"])
+                return pd.DataFrame(data=yvp_dicts, index=index)
         return pd.DataFrame()
 
-    def get_company_projected_targets(self, company_ids: List[str]) -> pd.DataFrame:
+    def get_company_projected_targets(self, company_ids: List[str], year=None) -> pd.DataFrame:
```

Comment:
we have `year` as an argument, but don't use it

Reply:
The test code uses it, when it looks for projected targets for the year 2025.

Reply:
How does unit test use it, if function under test makes no use of it?
Whichever value unit test provides - it will have no effect on the result.
Which meaning does it bring to the test case?

Reply:
Maybe outdated code?  I see:

```
        if projected_trajectories:
            if year:
                company_ids, scopes, values = list(map(list, zip(*[(pt[0], pt[1], [yvp.value for yvp in pt[2] if yvp.year==year][0]) for pt in projected_trajectories])))
                index=pd.MultiIndex.from_tuples(zip(company_ids, scopes), names=["company_id", "scope"])
                return pd.Series(data=values, index=index)
            else:
                company_ids, scopes, yvp_dicts = list(map(list, zip(*[(pt[0], pt[1], {yvp.year:yvp.value for yvp in pt[2]}) for pt in projected_trajectories])))
                index=pd.MultiIndex.from_tuples(zip(company_ids, scopes), names=["company_id", "scope"])
		return pd.DataFrame(data=yvp_dicts, index=index)

```

Which uses year both in the conditional and in the zip construct if year is not None.  (It should be changed to `if year is not None`).

Reply:
maybe you talk about method `get_company_projected_trajectories` - for which i agree, `year` is used there as in your snapshot.

i talk about method `get_company_projected_targets` - where i see `year` as not used

Reply:
Ah...good point.  Clearly we could write a test case as exists for get_company_projected_trajectories that would trigger the need to correctly handle the specific year case.  So we should write that test case and make sure it works.

Reply:
I just pushed changes with comments in the code and a test case to validate.  And I screwed up the DCO (so easy to do with GitHub desktop...grr!).

Reply:
Changes have been pushed (with unit test0 to use the year parameter when supplied.